starboard/build/install/install_target.gni

```diff
@@ -15,6 +15,8 @@
 import("//starboard/build/config/install.gni")
 
 template("install_target") {
+  not_needed(invoker, [ "shared_library_target_name" ])
```

Comment:
@arjungm please make a follow up bug to clean these up and ideally add TODOs inline as well

Reply:
Feel free to reuse b/305091479 if you think it's appropriate